scripts/mcp/antigravity_delegate.py

```
@@ -121,55 +121,53 @@
+        fcntl.ioctl(child_fd, termios.TIOCSCTTY, 0)
+        os.dup2(child_fd, 0)
+        os.dup2(child_fd, 1)
+        os.dup2(child_fd, 2)
+        if child_fd > 2:
+            os.close(child_fd)
+        os.close(parent_fd)  # R3: close parent fd in child before exec
+        # Intentional: no-shell argv-list invocation is the required injection-safe
+        # spawn pattern (fw-adr-0027 §3 R1; CWE-88; security-engineer-approved).
+        # Bandit B606 flags os.execvp as "start_process_with_no_shell" — that is
+        # precisely the point: no shell means no shell injection.  False positive.
+        os.execvp(argv[0], argv)  # nosec B606
```

Ruff (0.15.15)

[error] Starting a process without a shell

(S606)